test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +114,65 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [Fact]
+        public void StateCanBeSetByProcessor()
+        {
+            var exportedItems = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(exportedItems, Field.State));
```

Comment:
this doesn't seem to be testing the actual pipeline? Could we use InMemoryExporter, and RedactionProcessr, and validate that InMemoryExporter receives the modified LogRecord?